geode-core/src/main/java/org/apache/geode/management/internal/configuration/converters/ClassNameConverter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.configuration.converters;
+
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.geode.cache.configuration.DeclarableType;
+import org.apache.geode.cache.configuration.ParameterType;
+import org.apache.geode.management.configuration.ClassName;
+
+/**
+ * this converts between ClassName and DeclarableType (xml domain object). Only string type
+ * parameter in the DeclarableType will be converted as initialization properties.
+ */
+public class ClassNameConverter extends ConfigurationConverter<ClassName, DeclarableType> {
+  @Override
+  protected ClassName fromNonNullXmlObject(DeclarableType xmlObject) {
+    Properties properties = new Properties();
+    List<ParameterType> parameters = xmlObject.getParameters();
+    for (ParameterType parameter : parameters) {
+      // ClassName only supports string type values
+      if (parameter.getString() != null) {
+        properties.setProperty(parameter.getName(), parameter.getString());
+      }
+    }
```

Comment:
Lines 33-39 could be written in functional style (preferred):
```java
xmlObject.getParameters().stream()
        .filter(pt -> pt.getString() != null)
        .forEach(p -> properties.setProperty(p.getName(), p.getString()));
```

Reply:
nice trick